File: csharp-union-reader-writer-example/csharp-union-reader-writer-example/DevAppConfiguration.cs
using System;
using System.Collections.Generic;

namespace Union;

public class DevAppConfiguration : IAppConfiguration
{
    private Type _type;

    private DevAppConfiguration(Type type)
    {
        _type = type;
    }

    public static DevAppConfiguration InitReader()
    {
        return new DevAppConfiguration(Type.READER);
    }

    public static DevAppConfiguration InitWriter()
    {
        return new DevAppConfiguration(Type.WRITER);
    }

    public string GetAuthProviderUrl()
    {
        return "https://users.intellicess.com";
    }

    public string GetAuthProviderRealm()
    {
        return "public-test";
    }

    public string GetAuthClientId()
    {
        return "example-app";
    }

    public string GetAuthUserUsername()
    {
        if (_type == Type.READER)
        {
            return ""; // TO FILL: reader username
        }
        else if (_type == Type.WRITER)
        {
            return ""; // TO FILL: writer username
        }
        throw new Exception("Type='" + _type + "' and is not recognized");
    }

    public string GetAuthUserPassword()
    {
        if (_type == Type.READER)
        {
            return ""; // TO FILL: reader password
        }
        else if (_type == Type.WRITER)
        {
            return ""; // TO FILL: writer password
        }
        throw new Exception("Type='" + _type + "' and is not recognized");
    }

    public string GetUnionUrl()
    {
        return "https://theunion.cloud";
    }

    public string GetUnionClient()
    {
        return "public-test";
    }

    public string GetUnionRegion()
    {
        return "texas";
    }

    public string GetUnionAsset()
    {
        return "well1";
    }

    public string GetUnionFolder()
    {
        return "shared-data";
    }

    public string GetLocalWorkingDirectory()
    {
        if (_type == Type.READER)
        {
            return "../reader-folder";
        }
        else if (_type == Type.WRITER)
        {
            return "../writer-folder";
        }
        throw new Exception("Type='" + _type + "' and is not recognized");
    }

    private enum Type
    {
        READER, WRITER
    }
}

File: java-union-reader-writer-example/src/main/java/com/intellicess/union/java_union_reader_writer_example/config/DevAppConfiguration.java
package com.intellicess.union.java_union_reader_writer_example.config;

import com.intellicess.union.java_union_reader_writer_example.utils.AppConfiguration;

public class DevAppConfiguration implements AppConfiguration {

    private final Type type;

    private DevAppConfiguration(Type type) {
        this.type = type;
    }

    public static DevAppConfiguration initReader() {
        return new DevAppConfiguration(Type.READER);
    }

    public static DevAppConfiguration initWriter() {
        return new DevAppConfiguration(Type.WRITER);
    }

    @Override
    public String getAuthProviderUrl() {
        return "https://users.intellicess.com";
    }

    @Override
    public String getAuthProviderRealm() {
        return "public-test";
    }

    @Override
    public String getAuthClientId() {
        return "example-app";
    }

    @Override
    public String getAuthUserUsername() {
        if (type == Type.READER) {
            return ""; // TO FILL: reader username
        } else if (type == Type.WRITER) {
            return ""; // TO FILL: writer username
        }
        throw new RuntimeException("Type='" + type + "' and is not recognized");
    }

    @Override
    public String getAuthUserPassword() {
        if (type == Type.READER) {
            return ""; // TO FILL: reader password
        } else if (type == Type.WRITER) {
            return ""; // TO FILL: writer password
        }
        throw new RuntimeException("Type='" + type + "' and is not recognized");
    }

    @Override
    public String getUnionUrl() {
        return "https://theunion.cloud";
    }

    @Override
    public String getUnionClient() {
        return "public-test";
    }

    @Override
    public String getUnionRegion() {
        return "texas";
    }

    @Override
    public String getUnionAsset() {
        return "well1";
    }

    @Override
    public String getUnionFolder() {
        return "shared-data";
    }

    @Override
    public String getLocalWorkingDirectory() {
        if (type == Type.READER) {
            return "./reader-folder";
        } else if (type == Type.WRITER) {
            return "./writer-folder";
        }
        throw new RuntimeException("Type='" + type + "' and is not recognized");
    }

    private enum Type {
        READER, WRITER
    }
}
